"Authentication" Link in Admin Portal Fails with 400 Bad Request After Migrating to Version 25

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

**Summary:** 
After performing a stepwise migration of a Keycloak server from version 21 to the latest version (25), the "Authentication" link in the admin portal causes a 400 Bad Request error. 

**Reproduction Rate:**

100% on upgraded instances (from version 21 to 25) with existing database data.
0% on fresh Keycloak instances that follow the same upgrade path without existing data.

**Environment:**

- Keycloak Version: 25, also tried nightly
- Previous Versions Tested: 21, 22, 23, 24 (no issues with the "Authentication" link)
- Deployment Environment:
- Production: AWS with ALB, Local Development: localhost:8080 (using start command)
- Configuration:
-- hostname-strict=false
-- proxy-headers=xforwarded and also tried proxy-headers-forwarded


![Error Message 1](https://github.com/user-attachments/assets/9c8b5e3e-7d55-4e3c-8f20-35b9085427c8)

**Investigation Findings:**
Please note that another [github issue ](https://github.com/keycloak/keycloak/issues/32531) noted a similar error and the solution was to make sure the "custom providers" were installed. Our custom providers have followed each upgrade from 21->22->23->24 and every time, the authentication link works. It's when we move to 25 or latest, this issue occurs. 

The issue is only reproducible in instances that have been upgraded from version 21 to 25 with existing database data.
Viewing the admin ui using chrome developer tools, I see the the "required-actions" endpoint returns a 400 - bad request almost immediately on clicking the "Authentication" link.
![Required Actions Resource](https://github.com/user-attachments/assets/3db80a62-0066-41ff-9ea5-ad6a19170347)
![Required Actions Response error](https://github.com/user-attachments/assets/1ba2945b-9441-441c-b634-c03349ba8852)

Here is the relevant snippet from the debug log:
`2024-09-03 17:23:30,078 DEBUG [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-14) Error response 400: com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke "org.keycloak.authentication.RequiredActionFactory.isConfigurable()" because "factory" is null
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._wrapAsIOE(DefaultSerializerProvider.java:531)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:504)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
        at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1574)
        at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1275)
        at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:1098)
        at io.quarkus.resteasy.reactive.jackson.runtime.serialisers.FullyFeaturedServerJacksonMessageBodyWriter.writeResponse(FullyFeaturedServerJacksonMessageBodyWriter.java:79)
        at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:217)
        at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:185)
        at org.jboss.resteasy.reactive.server.core.serialization.FixedEntityWriter.write(FixedEntityWriter.java:28)
        at org.jboss.resteasy.reactive.server.handlers.ResponseWriterHandler.handle(ResponseWriterHandler.java:34)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:147)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "org.keycloak.authentication.RequiredActionFactory.isConfigurable()" because "factory" is null
        at org.keycloak.admin.ui.rest.AuthenticationManagementResource.toRepresentation(AuthenticationManagementResource.java:144)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEachOrdered(ReferencePipeline.java:601)
        at com.fasterxml.jackson.datatype.jdk8.StreamSerializer.serialize(StreamSerializer.java:71)
        at com.fasterxml.jackson.datatype.jdk8.StreamSerializer.serialize(StreamSerializer.java:15)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502)
`
**Suggested Priority:** High - This issue affects the ability to manage authentication settings in the admin portal, which is a crucial functionality for maintaining security and access control.

**Acceptance Criteria:**

The "Authentication" link in the admin portal should work as expected after migrating from version 21 to version 25 with existing database data.
Ensure that the functionality is consistent across both upgraded and fresh instances.

### Version

nightly

### Regression

- [X] The issue is a regression

### Expected behavior

**Expected Behavior:** The "Authentication" link should function correctly and load the authentication management page without any errors, consistent with previous versions up to 24.

### Actual behavior

**Actual Behavior:** Clicking on the "Authentication" link results in a 400 Bad Request error as indicated in the browser's developer tools. This issue does not occur when using a fresh Keycloak instance deployed from version 21 to 25 following the same upgrade path. 


### How to Reproduce?

**Steps to Reproduce:**
Please Note: This issue is observed only after the migration to version 25 using our database/existnig data and does not occur in a fresh deployment following the same upgrade path. The problem appears to be linked to the upgraded database configuration.
1. Start with an upgraded Keycloak instance, initially on version 21, with existing database data.
2. Migrate the Keycloak server stepwise through major versions (21 -> 22 -> 23 -> 24 -> 25).
3. Configure Keycloak with the following settings:
- hostname-strict=false
- tried both proxy-headers=xforwarded and proxy-headers=forwarded in different test cases

4. Access the admin portal and navigate to various menu items within any realm to verify general functionality.
5. Click on the "Authentication" link in the admin portal.


### Anything else?

_No response_

--- a/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/AuthenticationManagementResource.java
+++ b/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/AuthenticationManagementResource.java
@@ -141,7 +141,9 @@ public class AuthenticationManagementResource extends RoleMappingResource {
         rep.setConfig(model.getConfig());
 
         RequiredActionFactory factory = (RequiredActionFactory)session.getKeycloakSessionFactory().getProviderFactory(RequiredActionProvider.class, model.getProviderId());
-        rep.setConfigurable(factory.isConfigurable());
+        if (factory != null) {
+            rep.setConfigurable(factory.isConfigurable());
+        }
 
         return rep;
     }

--- a/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/AuthenticationManagementResource.java
+++ b/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/AuthenticationManagementResource.java
@@ -18,6 +18,7 @@ import org.eclipse.microprofile.openapi.annotations.media.Content;
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
 import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.logging.Logger;
 import org.jboss.resteasy.reactive.NoCache;
 import org.keycloak.admin.ui.rest.model.Authentication;
 import org.keycloak.admin.ui.rest.model.AuthenticationMapper;
@@ -37,6 +38,9 @@ import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluato
 
 
 public class AuthenticationManagementResource extends RoleMappingResource {
+
+    private static final Logger logger = Logger.getLogger(AuthenticationManagementResource.class);
+
     public AuthenticationManagementResource(KeycloakSession session, RealmModel realm, AdminPermissionEvaluator auth) {
         super(session, realm, auth);
     }
@@ -143,6 +147,9 @@ public class AuthenticationManagementResource extends RoleMappingResource {
         RequiredActionFactory factory = (RequiredActionFactory)session.getKeycloakSessionFactory().getProviderFactory(RequiredActionProvider.class, model.getProviderId());
         if (factory != null) {
             rep.setConfigurable(factory.isConfigurable());
+        } else {
+            logger.warnv("Detected RequiredAction with missing provider. realm={0}, alias={1}, providerId={2}",
+                    realm.getName(), model.getAlias(), model.getProviderId());
         }
 
         return rep;

